framework/src/problems/FEProblemBase.C

```diff
@@ -4118,122 +4118,136 @@ FEProblemBase::computeUserObjects(const ExecFlagType & type, const Moose::AuxGro
 void
 FEProblemBase::computeUserObjectsInternal(const ExecFlagType & type,
                                           const Moose::AuxGroup & group,
-                                          TheWarehouse::Query & query)
+                                          TheWarehouse::Query & primary_query)
 {
   // Add group to query
   if (group == Moose::PRE_IC)
-    query.condition<AttribPreIC>(true);
+    primary_query.condition<AttribPreIC>(true);
   else if (group == Moose::PRE_AUX)
-    query.condition<AttribPreAux>(type);
+    primary_query.condition<AttribPreAux>(type);
   else if (group == Moose::POST_AUX)
-    query.condition<AttribPostAux>(type);
+    primary_query.condition<AttribPostAux>(type);
 
-  std::vector<GeneralUserObject *> genobjs;
-  query.clone().condition<AttribInterfaces>(Interfaces::GeneralUserObject).queryInto(genobjs);
+  // query everything first to obtain a list of execution groups
+  std::vector<UserObject *> uos;
+  primary_query.clone().condition<AttribInterfaces>(Interfaces::UserObject).queryInto(uos);
```

Comment:
why do we clone the queries again?

Reply:
Maybe because attaching a condition modifies the query object? I'll see if I can figure that out.

Reply:
^ that's exactly it